crates/primitives/src/chain/spec.rs

```diff
@@ -450,7 +481,7 @@ pub struct ChainSpec {
     pub deposit_contract: Option<DepositContract>,
 
     /// The parameters that configure how a block's base fee is computed
-    pub base_fee_params: BaseFeeParams,
+    pub base_fee_params: Vec<(Hardfork, BaseFeeParams)>,
```

Comment:
this should be a helper type.

ideally an enum with two variants, one with BaseFeeParams that represents the previous impl and another that's a wrapper type around `Vec<()>`